### ISSUE TITLE
Replace hack/ensure-*.sh scripts with `go install`

### ISSUE THREAD

**Issue by mboersma** [kubernetes-sigs/cluster-api-provider-azure#1849]
/kind feature

**Describe the solution you'd like**

In the `./hack` directory are scripts named `ensure-<something>.sh`. Some of these install Go-based tools, and they could perhaps be replaced entirely by invocations of `go install`.

**Anything else you would like to add:**

See [this comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1842#pullrequestreview-802891068) for more context.

**Environment:**

N/A

**Comment by jsturtevant** [comment]
related to https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1192

**Comment by Darshnadas** [comment]
Is this issue been taken? if not, I would like to help in it.

**Comment by CecileRobertMichon** [comment]
@Darshnadas go for it 🚀 

let us know if you need any help

**Comment by Darshnadas** [comment]
Thank you @CecileRobertMichon, I would like to take some more insights into this issue to completely understand.

**Comment by sayantani11** [comment]
Will all the ensure scripts go in one PR? So guess even I can help

**Comment by CecileRobertMichon** [comment]
/remove-lifecycle stale
/unassign @Darshnadas

Please feel free to reassign if you pick this up again

**Comment by nawazkh** [comment]
TL;DR: I was wondering if we could try making a Makefile recipe to install `az` binary or leave `ensure-azcli.sh` as is if it is only used in the CI.

----
There are five more `ensure-*.sh` scripts left after the PRs for https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1192 were merged to the repo.

Brief summary of the five scripts:
- `ensure-azcli.sh`: Installs `az` binary (assuming the underlying OS is a linux distribution) and logs into `az` account. 
- `ensure-acr-login.sh`: Uses `ensure-azcli.sh` to log into `az` account and then logs into azure container registry.
- `ensure-acr-cleanup-schedule.sh`: creates a cleanup task (to remove untagged images?)
- `ensure-go.sh`: checks if installed go (from the `$PATH`) is `>=` minimum version `1.17.0`
- `ensure-tags.sh`: is exporting `BUILD_PROVENANCE` tags with the required repo_name, repo_owner and other PR details. 

Among them, `ensure-azcli.sh` is the one _installing_ a binary (`az`). While there are multiple ways of installing `az` in a user's env, there doesn't seem to be a `go install` way to do it. (Referring to https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)

`ensure-azcli.sh` in turn gets called by `ci-build-kubernetes.sh` and I do not see any mention of `ensure-azcli.sh`'s standalone use.

**Comment by mboersma** [comment]
I think it's ok to leave `ensure-azcli.sh` and the other remaining scripts as is. This issue is scoped to Go-based tools, and you completed that @nawazkh.
